components/org.wso2.carbon.identity.application.authenticator.facebook/src/main/java/org/wso2/carbon/identity/application/authenticator/facebook/FacebookAuthenticator.java

```diff
@@ -334,22 +339,55 @@ protected void setSubject(AuthenticationContext context, Map<String, Object> jso
         return jsonObject;
     }
 
+    /**
+     * Check whether there is a dedicated dialect defined for this connector.
+     *
+     * @param context The context
+     * @return true if claim is defined.
+     */
+    private boolean isClaimDialectExist(AuthenticationContext context) {
+        if (externalClaims == null) {
+            ClaimMetadataManagementServiceImpl claimMetadataService = new ClaimMetadataManagementServiceImpl();
+            try {
+                externalClaims = claimMetadataService.getExternalClaims(
```

Comment:
Better to get dialect list and check for the existence of FB dialect. 

Don't store it locally.. Dialect list should maintained tenant wise, and there will be caching issues when storing locally